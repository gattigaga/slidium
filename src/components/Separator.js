import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Line = styled.div`
  width: ${({ orientation }) =>
    orientation === "horizontal" ? "100%" : "0px"};
  height: ${({ orientation }) =>
    orientation === "horizontal" ? "0px" : "100%"};
  margin: ${({ orientation }) =>
    orientation === "horizontal" ? "4px 0px" : "0px 4px"};
  box-sizing: border-box;

  ${({ orientation }) =>
    orientation === "horizontal"
      ? "border-top"
      : "border-right"} : 1px solid #ccc;
`;

const Separator = ({ orientation }) => <Line orientation={orientation} />;

Separator.propTypes = {
  orientation: PropTypes.oneOf(["horizontal", "vertical"])
};

Separator.defaultProps = {
  orientation: "horizontal"
};

export default Separator;
