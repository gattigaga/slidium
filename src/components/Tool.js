import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Icon from "components/Icon";
import Tooltip from "components/Tooltip";

const StyledIcon = styled(Icon)`
  color: ${({ isActive }) => (isActive ? "#222" : "#888")};
  font-size: 20px;
  margin: auto;
`;

const StyledTooltip = styled(Tooltip)``;

const Container = styled.div`
  width: 32px;
  height: 32px;
  box-sizing: border-box;
  border-radius: 4px;
  background: ${({ isActive }) => (isActive ? "#ddd" : "white")};
  margin: 4px;
  display: flex;
  cursor: pointer;
  user-select: none;
  position: relative;

  &:hover {
    background: ${({ isActive }) => (isActive ? "#ddd" : "#eee")};
  }

  &:hover ${StyledIcon} {
    color: #222;
  }

  &:hover ${StyledTooltip} {
    opacity: 1;
  }
`;

const Tool = ({ icon, tooltip, isActive, onClick }) => (
  <Container onClick={onClick} isActive={isActive}>
    {tooltip && <StyledTooltip text={tooltip} />}
    <StyledIcon name={icon} isActive={isActive} />
  </Container>
);

Tool.propTypes = {
  icon: PropTypes.string.isRequired,
  tooltip: PropTypes.string,
  onClick: PropTypes.func,
  isActive: PropTypes.bool
};

Tool.defaultProps = {
  isActive: false
};

export default Tool;
