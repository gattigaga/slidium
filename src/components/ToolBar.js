import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Icon from "components/Icon";

const Container = styled.div`
  width: 100%;
  height: 40px;
  box-sizing: border-box;
  background: white;
  display: flex;
  padding: 0px 8px;
  border-bottom: 1px solid #ccc;
  align-items: center;
  justify-content: space-between;
`;

const ToolContainer = styled.div`
  width: auto;
  height: 100%;
  box-sizing: border-box;
  display: flex;
  align-items: center;
`;

const Toggle = styled(Icon)`
  color: #888;
  cursor: pointer;
  padding: 4px;
  user-select: none;

  &:hover {
    color: black;
  }
`;

Toggle.displayName = "Toggle";

const ToolBar = ({ children, isExpanded, onClickToggle }) => (
  <Container>
    <ToolContainer>{children}</ToolContainer>
    <Toggle
      name={isExpanded ? "expand_less" : "expand_more"}
      onClick={onClickToggle}
    />
  </Container>
);

ToolBar.propTypes = {
  children: PropTypes.node,
  isExpanded: PropTypes.bool,
  onClickToggle: PropTypes.func
};

ToolBar.defaultProps = {
  isExpanded: true
};

export default ToolBar;
