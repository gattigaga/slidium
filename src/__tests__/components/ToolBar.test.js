import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import ToolBar from "components/ToolBar";
import Tool from "components/Tool";
import Separator from "components/Separator";

let wrapper, onClickToggle;

beforeEach(() => {
  onClickToggle = jest.fn();
  wrapper = shallow(<ToolBar onClickToggle={onClickToggle} />);
});

describe("ToolBar", () => {
  it("renders with empty tool", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with some tools", () => {
    wrapper = shallow(
      <ToolBar onClickToggle={onClickToggle}>
        <Tool icon="undo" />
        <Tool icon="redo" />
        <Separator orientation="vertical" />
      </ToolBar>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders not expanded", () => {
    wrapper.setProps({
      isExpanded: false
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClickToggle'", () => {
    expect(onClickToggle.mock.calls.length).toEqual(0);
    wrapper.find("Toggle").simulate("click");
    expect(onClickToggle.mock.calls.length).toEqual(1);
  });
});
