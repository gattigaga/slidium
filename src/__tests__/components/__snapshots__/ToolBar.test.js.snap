// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar renders not expanded 1`] = `
<styled.div>
  <styled.div />
  <Toggle
    name="expand_more"
    onClick={[Function]}
  />
</styled.div>
`;

exports[`ToolBar renders with empty tool 1`] = `
<styled.div>
  <styled.div />
  <Toggle
    name="expand_less"
    onClick={[Function]}
  />
</styled.div>
`;

exports[`ToolBar renders with some tools 1`] = `
<styled.div>
  <styled.div>
    <Tool
      icon="undo"
    />
    <Tool
      icon="redo"
    />
    <Separator
      orientation="vertical"
    />
  </styled.div>
  <Toggle
    name="expand_less"
    onClick={[Function]}
  />
</styled.div>
`;
